src/visual.ts

```diff
@@ -194,32 +206,43 @@ export class TableHeatMap implements IVisual {
         });
 
         dataView.categorical.categories[0].values.forEach((categoryX, indexX) => {
-            dataView.categorical.values.forEach((categoryY) => {
-                const categoryYFormatter = valueFormatter.create({
-                    format: categoryY.source.format,
-                    value: dataView.categorical.values[0].values[0]
-                });
-                const value = categoryY.values[indexX];
-                dataPoints.push({
-                    categoryX: categoryX,
-                    categoryY: categoryY.source.displayName,
-                    value: value,
-                    valueStr: categoryYFormatter.format(value),
-                    tooltipInfo: [{
-                        displayName: `Category`,
-                        value: (categoryX || "").toString()
-                    },
-                    {
-                        displayName: `Y`,
-                        value: (categoryY.source.displayName || "").toString()
-                    },
-                    {
-                        displayName: `Value`,
-                        value: categoryYFormatter.format(value)
-                    }]
-                });
-            });
+        dataView.categorical.values.forEach((categoryY, indexY) => {
```

Comment:
Is that a mandatory change?

Reply:
The inner forEach loop is crucial for accurately processing data in the "Y" category, generating the necessary categoryYFormatter and categorySelectionId. Removing it could lead to inaccurate data processing and potential code errors.